homeassistant/components/tts/__init__.py

```diff
@@ -437,6 +427,19 @@ def async_get_supported_voices(self, language: str) -> list[Voice] | None:
     async def async_internal_added_to_hass(self) -> None:
         """Call when the entity is added to hass."""
         await super().async_internal_added_to_hass()
+        if not (
+            hasattr(self, "_attr_default_language") or hasattr(self, "default_language")
```

Comment:
Won't the property make the hasattr check return true?

Reply:
The whole `if` is only true when neither the attribute nor the property is set.

Reply:
Right, I mean the property defined by the parent.

Reply:
The property of the parent checks for the `_attr` attribute and if it doesn't exists, an `AttributeError` is thrown. So actually we just need to access the properties and check if an AttributeError is thrown, which is only when the subclass neither has the properties nor the attribute variables set.